### DISCUSSION TITLE
Channel back up restore issue

### DISCUSSION THREAD

**Discussion by saubyk** [Ride-The-Lightning/RTL#1125] [Troubleshooting Questions] [answered]
Query reference:
https://github.com/Ride-The-Lightning/RTL/issues/614#issuecomment-1295871081

Hi @lazyrudi, creating a discussion thread here to investigate your query further

**Comment by saubyk** [comment]
Before we can advise you any further, please answer a few basic questions:
- What version of LND are you running?
- What version of RTL are you running?
- Is `PinselNeu` a new node all together or did you restore it with the seed phrase of `Pinsel`?

**Comment by lazyrudi** [comment]
Where can I find the LND version number? Is it the RaspiBlitz V.1.8.0 or what number are you looking for?

The RTL version is 0.12.3-beta

As I could not fix the problems with my old installation I decided to do a complete new installation with a new name, new PWDs and new seed phrase. This for me was the only way to copy the Blockchain from an other node I am running, without waiting 3 weeks for installing the new blockchain from scratch. If there is any way to reinstall the old node and install the channel backup without problems and waiting 3 weeks for the blockchain sync please let me know. Then I will reinstall the old node with the old seed phrase.

**Comment by saubyk** [comment, marked as answer]
So regarding restoring funds from channel backups, there are a few basic things you should understand:
1. You *must* have the node restored from the same seed as the node which you backed up the channels from. You cannot restore the funds from one node to another
2. The *restoration* is of the funds, not the channel. So when you are able to complete the restore function, it will force close all your old channels and you will get your funds back *onchain* minus the mining fee and after a delay

Regarding the wait for 3 weeks etc. for IBD, I don't have any direct solutions that I can recommend. May be if you have access to the old node's drive you can try copying the block and index data. But it's a technically complex operation and since you're using Raspiblitz, I am not sure of its folder structure to recommend the exact actions. You can try reaching out to them, in case they have a simple solution for you.
Also, depending on the amount of funds you're trying to restore, waiting for 3 weeks may not be so bad, if you want to keep things simple.

To reiterate, the basic things to understand regarding the channel restoration is:
- first, restore the node from the original seed
- second, attempt restoration of channel funds
You cannot do the second without the first

**Comment by lazyrudi** [comment]
Ok got it. Now I understand the reason for the problem I am facing with the backup. Tomorrow I will Setup the node again with the old seed phrase. Thanks for your help!!

**Comment by lazyrudi** [comment]
After running in the 0.15.4 synchronizing Bug, I finally managed to restore the node with the old seed, old name (Pinsel) and old USB backup file. But I still fail to install the RTL channel Backup. I copied the „channel-all-5.bak file to the directory mentioned in the RTL-Backup-Restore function (/mnt/hdd/app-data/rtl/RTL-SCB-backup-Pinsel/restore). The copied file looks ok for me.

￼

The all:5 file is also shown in the restore function, but when I try to execute the function nothing happen. The channels are not restored but there is also no error message.

￼

This are the channels / values which should show up after the backup:

￼

**Comment by ShahanaFarooqui** [comment]
@lazyrudi Unfortunately, I am unable to see your backup file and structure through your comments. So I am adding an step by step image guide for your reference.

1: Your backup file has old RTL naming convention. With newer RTL versions, it should be named as 'channel-all.bak'. So please rename your file from `channel-all-5.bak` to `channel-all.bak` as shown in the image below:
![Screenshot from 2022-11-07 08-12-25](https://user-images.githubusercontent.com/23240721/200359243-3bd897c7-03c7-4ad8-91f3-ec1ad2af18f8.png)


2: Once that is done, create a folder named `restore` at the same location and copy your backup file in this folder:
![Screenshot from 2022-11-07 08-09-35](https://user-images.githubusercontent.com/23240721/200358860-91b9f544-faab-4ce6-a7cf-5ab8d954fac6.png)

3: Restart RTL server (if needed). It should enable the `Restore All` button. Please check that `restore folder location` is as expected.
![Screenshot from 2022-11-07 07-58-11](https://user-images.githubusercontent.com/23240721/200356477-2c9ca402-ae86-4a0e-888d-ea484f0bfbec.png)

4: The successful channel backup will:
4.1: send a message to the UI like:
![Screenshot from 2022-11-07 08-26-57](https://user-images.githubusercontent.com/23240721/200362663-5cceced9-5103-4af0-befc-52e8a01c232e.png)

4.2: And rename the backup file with `.restored` extention:
![Screenshot from 2022-11-07 08-29-04](https://user-images.githubusercontent.com/23240721/200363098-2252dbb7-5430-4ed2-953d-b4bc4b99e322.png)

Please post your backup file, before & after folder structure and console messages (after masking sensitive info), if it still does not resolve your issue.

**Comment by lazyrudi** [comment]
Hi,

I copied the old backup file to the folder as mentioned. Then I renamed the file to Channel-all.bak.

￼


Than I started RTL and found the „Restore All“ button

￼

Then I ran the backup reset. The file was renamed by the system to channel-all.restored

￼

But there still is no channel 

￼

**Comment by lazyrudi** [comment]
Sorry I missed to send the backup files. As I have no idea what are the critical data I just placed a box over the text.

Before use

￼

After use

￼


====

Hi,

I copied the old backup file to the folder as mentioned. Then I renamed the file to Channel-all.bak.

￼


Than I started RTL and found the „Restore All“ button

￼

Then I ran the backup reset. The file was renamed by the system to channel-all.restored

￼

But there still is no channel 

￼

**Reply by saubyk** [reply]
Hi @lazyrudi please read the comments again:
https://github.com/Ride-The-Lightning/RTL/discussions/1125#discussioncomment-4013316

Note that in the recovery from the backups, you will get all your old channels 'force closed' and the funds on the local balance of each channel will be returned to you on-chain. The channels are never 'restored' in this process.
You should see a bunch of channel close requests on the `Pending` channels tab.

**Comment by lazyrudi** [comment]
Thanks for the info. It took a while till the force close was processed. That’s why I didn’t see the money. Now all the Sats are back. Thanks very, very much for your help 👍🏻🫶🏼

**Comment by lazyrudi** [comment]
Why didn’t I get all the money back I put in my Channels?

￼

**Reply by saubyk** [reply]
No clear what you mean by "all the money"? Here is what you should track:
1. Are **all** the channel force closed?
2. Did you get the funds on the **local balance** for each channel closed?

If the answer to the above is yes, you got all the money that was yours

**Comment by lazyrudi** [comment]
My Node got an issue. That’s why I had to set it up completely new. Installing the channel backup processed an „force close“. Opening a channel with „bfx-Ind1“ required a min. Capacity of 4 Mio. Sats. After force close I only got a payback of 2,4 Mio. Sats. Where did the rest go? >:﻿
No clear what you mean by "all the money"? Here is what you should track:

Are the channel force closed?
Did you get the funds on the local balance for each channel closed?

If the answer to the above is yes, you got all the money that was yours

**Reply by saubyk** [reply]
Please read my previous comment again. The funds that you will get back is the **local balance** only, not the entire channel capacity. Because what you have on the local side only is **your money**, not the entire capacity. The remote balance goes to your channel counter part.

The channel that you're referring to may have a total capacity of 4 Mil, but my guess is only 2.4 Mil was on the **local balance**. No way for me to confirm this, it's something you should have known or seen when your node was active and online.

**Comment by lazyrudi** [comment]
Ok I got it. It really was the local balance. I’m still struggling with the channel financing and balancing. 🙈For opening the channel I had to invest 4 Mio. Sats, but how did 1,5 Mio. Sats move to the counter part? I didn’t sent any? I know that the channel should have 50/50 in- & outbound liquidity. But I still don’t understand how to achieve that and rebalance the channels. 🤷🏼

**Reply by saubyk** [reply]
> but how did 1,5 Mio. Sats move to the counter part?

Unfortunately, I cannot answer this question without knowing what activity was going on your node.
There are so many factors influencing it:
- How many other channels you had open?
- What kind of payments you made and to whom?
- What kind of payment forwarding happened on your node?

bfx-lnd1 is a big node on the network, which can attract a lot of traffic, so your node may have forwarded some payments to them, even if you did not make any payments directly. In case the forwarding happened, the same amount of funds would have shifted from remote to local on your other channel (where the payment came from). So the 1.5M funds which moved to the remote on this channel, would have resulted in 1.5M moving to your local on another (which you would have gotten when that channel closed).

All in all, you don't "lose" any funds. All the funds on your local side (on all the channels) is returned to you, which is all you own on your lightning node.

If this is still not clear, I recommend watching some videos or reading a bit more to improve your understanding of how lightning works. Thanks.
